libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/factory/RoadObjectFactory.kt

```diff
@@ -37,6 +37,27 @@ object RoadObjectFactory {
             }
     }
 
+    fun List<UpcomingRoadObject>.getUpdatedObjectsAhead(
+        status: com.mapbox.navigator.NavigationStatus
+    ): List<UpcomingRoadObject> {
+        val idToDistanceRemaining = status.upcomingRouteAlerts.associate {
+            it.roadObject.id to it.distanceToStart
+        }
+        val updateObjects = mutableListOf<UpcomingRoadObject>()
+        forEach {
+            if (it.roadObject.id in idToDistanceRemaining.keys) {
+                updateObjects.add(
+                    buildUpcomingRoadObject(
+                        roadObject = it.roadObject, // reusing the old road object reference
```

Comment:
How does this reuse work? The map from which road object is taken was build from a fresh status a few lines above, it seems that every time you wrap fresh route objects, don't you?

Reply:
I see, thanks